recordHooks/employees/apiValidation.ts

```diff
@@ -0,0 +1,38 @@
+export async function checkApiForExistingWorkers(record, employees) {
+  try {
+    // Get the current value of the Applicant_ID field
+    let employeeId = record.get('employeeId');
+
+    console.log('Employee_Id:', employeeId); // Log the current value of Applicant_ID
+
+    // Check if the Applicant_ID matches an id from the API data
+    const matchingEmployee = employees.find((employee) => {
+      // Ensure both the applicantId and employee.id are of the same type before comparing
+      // If applicantId is a string, compare it to a string version of employee.id
+      if (typeof employeeId === 'string') {
+        return employee.employeeId === employeeId;
+      } else {
+        // If applicantId is not a string (assuming it's a number), compare it to a number version of employee.id
+        const employeeIdNum = parseInt(employee.id, 10);
+        return employeeIdNum === employeeId;
+      }
+    });
```

Comment:
Did something change in the dataset that causes this to be a string or an integer?

Reply:
@colinfrederickson A Q for you?

Reply:
hcm.show should only be returning strings

Reply:
Removed conditionals